### PR TITLE
New Election Reports Module

### PR DESCRIPTION
**Pull request by Ivan-Edan**
**Description:** This PR is all about the Newly Revised Module which is the Election Reports.
**Added Function/Changes:**

_**- Election Reports part**_

1. Can download PDF for the overall winners.
2. School Year Button is working.
3. Can show name, section, course, position and number of votes.

_**- Graph Part**_

1. The graph can show the name and number of votes for the candidate of every position
2. The button for position is working.

**_- Voters Turnout_**

1. The functions to show data is working

**Functions that wasn't done yet:**

- Need to add pdf for the overall sections of the module
- Need to add vote percentage and total on the pdf file
- Need to add Excel
- Need to fix the next and previous button
- Need to dynamically change the color of the bar graph and doughnut graph.
- Need to add abstain and picture in the bar graph.
- All of the function for the feedback and suggestion part are still in progress.
- The overall U.I and responsiveness for every device size is not yet integrated in and will continue to work.

**Sample output:**
image
image
image
image
image

### CODE REVIEW

File: src/fetch_candidates.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
+<?php
+session_start();
```
Comment (Carl-Tabuso) [inline src/fetch_candidates.php:2]: Remove this. `session-handler.php` is already included. Also, this is not a page but rather a script that works on the server side. Move this inside the `includes` folder.

File: src/fetch_candidates.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            $position_id = $_GET['position_id'];
+            $election_year = $_GET['election_year'];
+
+            // Prepare and execute query
+            $query = $connection->prepare("SELECT c.first_name, c.last_name, COUNT(v.vote_id) as vote_count
+                                        FROM candidate c
+                                        JOIN vote v ON c.candidate_id = v.candidate_id
+                                        WHERE c.position_id = ? AND c.election_year = ?
+                                        GROUP BY c.candidate_id
+                                        ORDER BY vote_count DESC");
```
Comment (Carl-Tabuso) [inline src/fetch_candidates.php:41]: Do you have a new column `election_year` in candidate table?

File: src/result-generation.php
Comment (Carl-Tabuso) [inline src/result-generation.php:1]: I don't understand anything with the code. I advise that next time, you segregate your SQL queries and javascript into different files. Please make your code more modularized. It'll also be hard to identify defects, improvements, and anything else in the future.

File: src/result-pdf.php
Comment (Carl-Tabuso) [inline src/result-pdf.php:1]: Please indent your codes properly. It is hard to read.

image


image

I suggest that you create a new folder—you can named it `report` or how you see fit. You can do this inside the `includes` folder.

**Review by Carl-Tabuso — COMMENTED** [review]
I can't access the reports page. I suspect that it was the missing `election_year` field in the candidate table.

image

File: src/fetch_candidates.php (hunk 2 above)
Comment (AraojoBenedict) [reply src/fetch_candidates.php:41]: [SQLResult.txt](https://github.com/user-attachments/files/15522171/SQLResult.txt)
SQL used in the result page

**Comment by Carl-Tabuso** [comment]
Keep in mind the resolution of other screen sizes. I've tested it on mine — 1280x1024 and this looks unpleasant.

image

I had to zoom out to 80% to make it look as it was supposed to.

Do we proceed with the merging of your branch? I will just issue new tickets linking to this PR so we can track the changes i've proposed. I need your reply on this ASAP @Ivan-Edan  @AraojoBenedict

**Comment by Ivan-Edan** [comment]
I think we can proceed with the merging. @Carl-Tabuso we can just fix the issues on the next PR.

**Comment by Carl-Tabuso** [comment]
> I think we can proceed with the merging. @Carl-Tabuso we can just fix the issues on the next PR.

Noted. I want to add to your A.Y. Election Year dropdown that you should use ajax to prevent reloading the page whenever an option is selected.

**Review by Carl-Tabuso — APPROVED** [review]
To issue new tickets.

**Pull request merged**